jobs/src/main/scala/dcos/metronome/model/JobRunSpec.scala

```diff
@@ -71,6 +73,8 @@ object JobRunSpec {
       def noSecretVolumesExists: Boolean = jobRunSpec.volumes.forall(v => !isSecretVolume(v))
       check(noSecretVolumesExists || jobRunSpec.ucr.isDefined, JobRunSpecMessages.fileBasedSecretsAreUcrOnly)
```

Comment:
I don't like this either... now that I'm reviewing again...

Reply:
can my mind... this one is perfect... it is a great short circuit... if noSecretVolumes... nothing left to check.